x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -642,7 +645,25 @@ const updateExistingDataStream = async ({
   esClient: ElasticsearchClient;
   logger: Logger;
 }) => {
+  const dataStreamResponse = await esClient.indices.getDataStream({ name: dataStreamName });
+  const currentBackingIndex =
+    dataStreamResponse.data_streams[0].indices[
+      dataStreamResponse.data_streams[0].indices.length - 1
+    ];
+
+  const currentBackingIndexData = (
+    await esClient.indices.get({
```

Comment:
I think you can do this in one query and take the last result, tested locally.

```
    const existingDs = await esClient.indices.get({
      index: dataStreamName,
    })

    // should take the latest write index
    const existingDsConfig = Object.values(existingDs)
    const currentBackingIndexConfig = existingDsConfig[existingDsConfig.length - 1]

    const currentIndexMode = currentBackingIndexConfig?.settings?.index?.mode;
```

Reply:
Thanks! Updated.